chromium_src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h

```diff
@@ -0,0 +1,35 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_TPCD_HEURISTICS_REDIRECT_HEURISTIC_TAB_HELPER_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_TPCD_HEURISTICS_REDIRECT_HEURISTIC_TAB_HELPER_H_
+
+#define RedirectHeuristicTabHelper RedirectHeuristicTabHelper_ChromiumImpl
+#define MaybeRecordRedirectHeuristic virtual MaybeRecordRedirectHeuristic
+
+#include "src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h"  // IWYU pragma: export
+#undef MaybeRecordRedirectHeuristic
+#undef RedirectHeuristicTabHelper
+
+// Disable RedirectHeuristicTabHelper functionality since we disable kDIPS, so
+// the upstream code would not work (and crash because it doesn't check that
+// dips_service_ can be a null).
```

Comment:
have we filed this as a bug with upstream?

Reply:
I am not sure how to reproduce it in Chrome to file an issue with upstream, tbh.

Reply:
isn't it enabled by a buildflag or feature flag? Disabling it in chrome should have the same effect?